Thanks, Alastair.  Copying the error below.  I'll admit it doesn't make a whole lot of sense to me.  Win32.cpp.obj is listed on the link line, and I can't see any reason why it wouldn't be exporting the right symbol.

```
[42/1307] Building CXX object stdlib\public\Threading\CMakeFiles\swiftThreading-windows-x86_64.dir\__\__\__\lib\Threading\Win32.cpp.obj
In file included from C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\lib\Threading\Win32.cpp:24:
In file included from C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\include\swift/Threading/Impl.h:67:
In file included from C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\include\swift/Threading/Impl/Win32.h:20:
C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\include\swift/Threading/Impl/Win32/Win32Defs.h(50,9): warning: 'RTL_SRWLOCK_INIT' macro redefined [-Wmacro-redefined]
#define RTL_SRWLOCK_INIT                                                       \
        ^
C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um\winnt.h(20709,9): note: previous definition is here
#define RTL_SRWLOCK_INIT {0}                            
        ^
1 warning generated.

...

[71/125] Building CXX object unittests\Threading\CMakeFiles\SwiftThreadingTests.dir\Thread.cpp.obj

...

[88/125] Linking CXX executable unittests\Threading\SwiftThreadingTests.exe
FAILED: unittests/Threading/SwiftThreadingTests.exe 
cmd.exe /C "cd . && "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe" -E vs_link_exe --intdir=unittests\Threading\CMakeFiles\SwiftThreadingTests.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100177~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100177~1.0\x64\mt.exe --manifests  -- C:\PROGRA~2\MICROS~1\2019\COMMUN~1\VC\Tools\MSVC\1429~1.300\bin\Hostx64\x64\link.exe /nologo stdlib\public\Threading\CMakeFiles\swiftThreading-windows-x86_64.dir\__\__\__\lib\Threading\C11.cpp.obj stdlib\public\Threading\CMakeFiles\swiftThreading-windows-x86_64.dir\__\__\__\lib\Threading\Linux.cpp.obj stdlib\public\Threading\CMakeFiles\swiftThreading-windows-x86_64.dir\__\__\__\lib\Threading\Pthreads.cpp.obj stdlib\public\Threading\CMakeFiles\swiftThreading-windows-x86_64.dir\__\__\__\lib\Threading\Win32.cpp.obj unittests\Threading\CMakeFiles\SwiftThreadingTests.dir\Mutex.cpp.obj unittests\Threading\CMakeFiles\SwiftThreadingTests.dir\ConditionVariable.cpp.obj unittests\Threading\CMakeFiles\SwiftThreadingTests.dir\Once.cpp.obj unittests\Threading\CMakeFiles\SwiftThreadingTests.dir\LinuxUlock.cpp.obj unittests\Threading\CMakeFiles\SwiftThreadingTests.dir\Fatal.cpp.obj unittests\Threading\CMakeFiles\SwiftThreadingTests.dir\Thread.cpp.obj  /out:unittests\Threading\SwiftThreadingTests.exe /implib:unittests\Threading\SwiftThreadingTests.lib /pdb:unittests\Threading\SwiftThreadingTests.pdb /version:0.0 /INCREMENTAL:NO /STACK:10000000 /INCREMENTAL:NO /subsystem:console -LIBPATH:T:\llvm\.\lib T:\llvm\lib\LLVMSupport.lib  T:\llvm\lib\llvm_gtest_main.lib  T:\llvm\lib\llvm_gtest.lib  lib\swift\windows\x86_64\swiftCore.lib  Synchronization.lib  T:\llvm\lib\LLVMSupport.lib  psapi.lib  shell32.lib  ole32.lib  uuid.lib  advapi32.lib  delayimp.lib  -delayload:shell32.dll  -delayload:ole32.dll  T:\llvm\lib\LLVMDemangle.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib && cd ."
LINK: command "C:\PROGRA~2\MICROS~1\2019\COMMUN~1\VC\Tools\MSVC\1429~1.300\bin\Hostx64\x64\link.exe /nologo stdlib\public\Threading\CMakeFiles\swiftThreading-windows-x86_64.dir\__\__\__\lib\Threading\C11.cpp.obj stdlib\public\Threading\CMakeFiles\swiftThreading-windows-x86_64.dir\__\__\__\lib\Threading\Linux.cpp.obj stdlib\public\Threading\CMakeFiles\swiftThreading-windows-x86_64.dir\__\__\__\lib\Threading\Pthreads.cpp.obj stdlib\public\Threading\CMakeFiles\swiftThreading-windows-x86_64.dir\__\__\__\lib\Threading\Win32.cpp.obj unittests\Threading\CMakeFiles\SwiftThreadingTests.dir\Mutex.cpp.obj unittests\Threading\CMakeFiles\SwiftThreadingTests.dir\ConditionVariable.cpp.obj unittests\Threading\CMakeFiles\SwiftThreadingTests.dir\Once.cpp.obj unittests\Threading\CMakeFiles\SwiftThreadingTests.dir\LinuxUlock.cpp.obj unittests\Threading\CMakeFiles\SwiftThreadingTests.dir\Fatal.cpp.obj unittests\Threading\CMakeFiles\SwiftThreadingTests.dir\Thread.cpp.obj /out:unittests\Threading\SwiftThreadingTests.exe /implib:unittests\Threading\SwiftThreadingTests.lib /pdb:unittests\Threading\SwiftThreadingTests.pdb /version:0.0 /INCREMENTAL:NO /STACK:10000000 /INCREMENTAL:NO /subsystem:console -LIBPATH:T:\llvm\.\lib T:\llvm\lib\LLVMSupport.lib T:\llvm\lib\llvm_gtest_main.lib T:\llvm\lib\llvm_gtest.lib lib\swift\windows\x86_64\swiftCore.lib Synchronization.lib T:\llvm\lib\LLVMSupport.lib psapi.lib shell32.lib ole32.lib uuid.lib advapi32.lib delayimp.lib -delayload:shell32.dll -delayload:ole32.dll T:\llvm\lib\LLVMDemangle.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:unittests\Threading\SwiftThreadingTests.exe.manifest" failed (exit code 1120) with the following output:
Thread.cpp.obj : error LNK2019: unresolved external symbol "class llvm::Optional<struct swift::threading_impl::stack_bounds> __cdecl swift::threading_impl::thread_get_current_stack_bounds(void)" (?thread_get_current_stack_bounds@threading_impl@swift@@YA?AV?$Optional@Ustack_bounds@threading_impl@swift@@@llvm@@XZ) referenced in function "private: virtual void __cdecl StackTest_stack_bounds_Test::TestBody(void)" (?TestBody@StackTest_stack_bounds_Test@@EEAAXXZ)
unittests\Threading\SwiftThreadingTests.exe : fatal error LNK1120: 1 unresolved externals
```

--- a/lib/Threading/Pthreads.cpp
+++ b/lib/Threading/Pthreads.cpp
@@ -101,18 +101,26 @@ swift::threading_impl::thread_get_current_stack_bounds() {
 #if defined(__FreeBSD__)
   if (pthread_attr_init(&attr))
     return {};
+
+  if (pthread_attr_get_np(pthread_self(), &attr)) {
+    pthread_attr_destroy(&attr);
+    return {};
+  }
+#elif defined(__linux__)
+  if (pthread_getattr_np(pthread_self(), &attr))
+    return {};
+#else
+  // We don't know how to get the thread attr for this platform.
+  return {};
 #endif
 
-  if (!pthread_getattr_np(pthread_self(), &attr)) {
-    if (!pthread_attr_getstack(&attr, &begin, &size)) {
-      stack_bounds result = { begin, (char *)begin + size };
-      pthread_attr_destroy(&attr);
-      return result;
-    }
-
+  if (!pthread_attr_getstack(&attr, &begin, &size)) {
+    stack_bounds result = { begin, (char *)begin + size };
     pthread_attr_destroy(&attr);
+    return result;
   }
 
+  pthread_attr_destroy(&attr);
   return {};
 }
 #endif

